From fb4d148004f28494e9fb5d2400a13308d07a7988 Mon Sep 17 00:00:00 2001
From: Andreas Krebbel <krebbel@linux.ibm.com>
Date: Thu, 7 Apr 2022 07:45:49 +0200
Subject: [PATCH 72/91] IBM zSystems: Add support for z16 as CPU name.

So far z16 was identified as arch14. After the machine has been
announced we can now add the real name.

(cherry picked from commit 69341966def7f6551bc4452684136831d6a6941c)
---
 gas/config/tc-s390.c | 2 +-
 gas/doc/as.texi      | 4 ++--
 gas/doc/c-s390.texi  | 4 ++--
 opcodes/s390-mkopc.c | 3 ++-
 4 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/gas/config/tc-s390.c b/gas/config/tc-s390.c
index f192a786f32..4af635beac3 100644
--- a/gas/config/tc-s390.c
+++ b/gas/config/tc-s390.c
@@ -293,7 +293,7 @@ s390_parse_cpu (const char *arg,
       S390_INSTR_FLAG_HTM | S390_INSTR_FLAG_VX },
     { STRING_COMMA_LEN ("z15"), STRING_COMMA_LEN ("arch13"),
       S390_INSTR_FLAG_HTM | S390_INSTR_FLAG_VX },
-    { STRING_COMMA_LEN (""), STRING_COMMA_LEN ("arch14"),
+    { STRING_COMMA_LEN ("z16"), STRING_COMMA_LEN ("arch14"),
       S390_INSTR_FLAG_HTM | S390_INSTR_FLAG_VX }
   };
   static struct
diff --git a/gas/doc/as.texi b/gas/doc/as.texi
index 1a13b5dda42..1986f70111d 100644
--- a/gas/doc/as.texi
+++ b/gas/doc/as.texi
@@ -1897,8 +1897,8 @@ Specify which s390 processor variant is the target, @samp{g5} (or
 @samp{arch3}), @samp{g6}, @samp{z900} (or @samp{arch5}), @samp{z990} (or
 @samp{arch6}), @samp{z9-109}, @samp{z9-ec} (or @samp{arch7}), @samp{z10} (or
 @samp{arch8}), @samp{z196} (or @samp{arch9}), @samp{zEC12} (or @samp{arch10}),
-@samp{z13} (or @samp{arch11}), @samp{z14} (or @samp{arch12}), or @samp{z15}
-(or @samp{arch13}).
+@samp{z13} (or @samp{arch11}), @samp{z14} (or @samp{arch12}), @samp{z15}
+(or @samp{arch13}), or @samp{z16} (or @samp{arch14}).
 @item -mregnames
 @itemx -mno-regnames
 Allow or disallow symbolic names for registers.
diff --git a/gas/doc/c-s390.texi b/gas/doc/c-s390.texi
index edd9737066d..c51b5a14ec8 100644
--- a/gas/doc/c-s390.texi
+++ b/gas/doc/c-s390.texi
@@ -18,7 +18,7 @@ and eleven chip levels. The architecture modes are the Enterprise System
 Architecture (ESA) and the newer z/Architecture mode. The chip levels
 are g5 (or arch3), g6, z900 (or arch5), z990 (or arch6), z9-109, z9-ec
 (or arch7), z10 (or arch8), z196 (or arch9), zEC12 (or arch10), z13
-(or arch11), z14 (or arch12), z15 (or arch13), or arch14.
+(or arch11), z14 (or arch12), z15 (or arch13), or z16 (or arch14).
 
 @menu
 * s390 Options::                Command-line Options.
@@ -72,7 +72,7 @@ are recognized:
 @code{z13} (or @code{arch11}),
 @code{z14} (or @code{arch12}),
 @code{z15} (or @code{arch13}), and
-@code{arch14}.
+@code{z16} (or @code{arch14}).
 
 Assembling an instruction that is not supported on the target
 processor results in an error message.
diff --git a/opcodes/s390-mkopc.c b/opcodes/s390-mkopc.c
index 9fb67847b80..c795895f48c 100644
--- a/opcodes/s390-mkopc.c
+++ b/opcodes/s390-mkopc.c
@@ -381,7 +381,8 @@ main (void)
       else if (strcmp (cpu_string, "z15") == 0
 	       || strcmp (cpu_string, "arch13") == 0)
 	min_cpu = S390_OPCODE_ARCH13;
-      else if (strcmp (cpu_string, "arch14") == 0)
+      else if (strcmp (cpu_string, "z16") == 0
+	       || strcmp (cpu_string, "arch14") == 0)
 	min_cpu = S390_OPCODE_ARCH14;
       else {
 	fprintf (stderr, "Couldn't parse cpu string %s\n", cpu_string);
-- 
2.35.1

